### PR TITLE
Add GH paths to doxygen CI

### PR DESCRIPTION
**Pull request by electrocucaracha**
#### Problem
Doxygen tool only [support some extension files](https://www.doxygen.nl/manual/starting.html#step1), therefore changes on other type files can trigger unnecessary CI executions.

#### Change overview
This change reduces the number of CI executions by restricting the file extensions supported by doxygen

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: .github/workflows/doxygen.yaml [hunk 1]
```diff
@@ -16,8 +16,113 @@ name: Doxygen
...
+            - '**.java'
+            - '**.cs'
+            - '**.d'
+            - '**.php'
+            - '**.php4'
+            - '**.php5'
+            - '**.inc'
+            - '**.phtml'
+            - '**.m'
+            - '**.M'
```
Comment (electrocucaracha) [inline .github/workflows/doxygen.yaml:53]: @andy31415 this is my initial attempt the use the [GH match filters](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#patterns-to-match-file-paths), unfortunately I don't see examples of grouping characters in the regex. 

```suggestion
            - '**.do[xc]'
            - '**.c[cs]?'
            - '**.cxx'
            - '**.cpp'
            - '**.c++'
            - '**.ii'
            - '**.ixx'
            - '**.ipp'
            - '**.i++'
            - '**.inl'
            - '**.[hH]'
            - '**.hh'
            - '**.HH'
            - '**.hxx'
            - '**.hpp'
            - '**.h++'
            - '**.mm'
            - '**.txt'
            - '**.[ido]dl'
            - '**.java'
            - '**.d'
            - '**.php[45]?'
            - '**.inc'
            - '**.phtml'
            - '**.[mM]'
```

**Comment by woody-apple** [comment]
Fast tracking, given > 3 days for review.

**Comment by woody-apple** [comment]
Fast tracking, given > 3 business days for review.

**Pull request merged**